stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,665 @@
+  }
+  explicit module netinet_in {
+    header "netinet/in.h"
+    export *
+  }
+  explicit module netinet_in6 {
+    header "netinet/in6.h"
+    export *
+  }
+  explicit module bits_ip_mreq_source {
+    header "bits/ip_mreq_source.h"
+    export *
+  }
+  explicit module bits_ip_msfilter {
+    header "bits/ip_msfilter.h"
+    export *
+  }
+  explicit module bits_in_addr {
+    header "bits/in_addr.h"
+    export *
+  }
+  explicit module linux_if {
+    header "linux/if.h"
+    export *
+  }
+  explicit module sys_socket {
+    header "sys/socket.h"
+    export *
+  }
+}
+
+module jni [system] {
+  header "jni.h"
+  export *
+}
+
+module langinfo [system] {
+  header "langinfo.h"
+  export *
+}
+
+module pthread [system] {
+  header "pthread.h"
+  export *
+}
+
+module pty [system] {
+  header "pty.h"
+  export *
+}
+
+module regex [system] {
+  header "regex.h"
+  export *
+}
+
+module resolv [system] {
+  header "resolv.h"
+  export *
+}
+
+module sched [system] {
+  header "sched.h"
+  export *
+}
+
+module search [system] {
+  header "search.h"
+  export *
+}
+
+module semaphore [system] {
+  header "semaphore.h"
+  export *
+}
+
+module spawn [system] {
+  header "spawn.h"
+  export *
+}
+
+module strings [system] {
+  header "strings.h"
+  export *
+}
+
+module sys [system] {
+  module syscall {
+    header "syscall.h"
+    export *
+  }
+  module sysexits {
+    header "sysexits.h"
+    export *
+  }
+  module syslog {
+    header "syslog.h"
+    export *
+  }
+}
+
+module tar [system] {
+  header "tar.h"
+  export *
+}
+
+module termio [system] {
+  module termio {
+    header "termio.h"
+    export *
+  }
+  module termios {
+    header "termios.h"
+    export *
+  }
+}
+
+module uconfig_local [system] {
+  header "uconfig_local.h"
+  export *
+}
+
+module ucontext [system] {
+  header "ucontext.h"
+  export *
+}
+
+module unistd [system] {
+  header "unistd.h"
+  export *
+  explicit module sys_unistd {
+    header "sys/unistd.h"
+    export *
+  }
+}
+
+module utmp [system] {
+  module utmp {
+    header "utmp.h"
+    export *
+  }
+  module utmpx {
+    header "utmpx.h"
+    export *
+  }
+}
+
+module wait [system] {
+  header "wait.h"
+  export *
+  explicit module sys_wait {
+    header "sys/wait.h"
+    export *
+  }
+}
+
+module xlocale [system] {
+  header "xlocale.h"
+  export *
+}
+
+// Additional modules in the 'android' subdirectory.
+module android_defs [system] {
+  module ndk_version {
+    header "android/ndk-version.h"
+    export *
+  }
+  module versioning {
+    header "android/versioning.h"
+    export *
+  }
+}
+
+module android_apis [system] {
+  module asset_manager_jni {
+    header "android/asset_manager_jni.h"
+    export *
+  }
+  module asset_manager {
+    header "android/asset_manager.h"
+    export *
+  }
+  module log {
+    header "android/log.h"
+    export *
+  }
+  module trace {
+    header "android/trace.h"
+    export *
+  }
+}
+
+// Additional modules in the 'bits' subdirectory.
```

Comment:
I'd started putting things like these into something like a "BionicFoundation" or "BionicCore" module for things that are 1) really _really_ low level and 2) probably get included by most programs already. Not sure if that applies to these though.